test/queries/ledger_test.go

```diff
@@ -498,3 +499,148 @@ func verifyTxProof(intermediateHashes [][]byte, receipt *types.TxReceipt, tx pro
 
 	return bytes.Equal(receipt.GetHeader().GetTxMerkelTreeRootHash(), currHash), nil
 }
+
+// Scenario:
+// HTTP GET "/ledger/proof/data/{blockId}/{dbname}/{key}?deleted={true|false}" gets proof for value associated with (dbname, key) in block blockId,
+// HTTP GET "/ledger/proof/data/{blockId}/{dbname}/{key}" gets proof for value associated with (dbname, key) in block blockId
+func TestLedgerDataProof(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	s := c.Servers[leaderIndex]
+
+	txID, rcpt1, _, err := s.WriteDataTx(t, worldstate.DefaultDBName, "key1", []byte("key1-data1"))
+	require.NoError(t, err)
+	require.NotNil(t, rcpt1)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt1.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt1.Header.ValidationInfo[rcpt1.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt1)
+
+	//get data proof
+	respEnv, err := s.GetDataProof(t, worldstate.DefaultDBName, "key1", "admin", rcpt1.GetHeader().GetBaseHeader().GetNumber(), false)
+	require.NoError(t, err)
+	require.NotNil(t, respEnv)
+
+	//verify data proof
+	valHash, err := calculateValueHash(worldstate.DefaultDBName, "key1", []byte("key1-data1"))
+	require.NoError(t, err)
+	require.NotNil(t, valHash)
+	ok, err := verifyDataProof(respEnv.GetResponse().GetPath(), valHash, rcpt1.GetHeader().GetTxMerkelTreeRootHash(), false)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	//change "key1" data
+	txID, rcpt2, _, err := s.WriteDataTx(t, worldstate.DefaultDBName, "key1", []byte("key1-data2"))
+	require.NoError(t, err)
+	require.NotNil(t, rcpt2)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt2.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt2.Header.ValidationInfo[rcpt2.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt2)
+
+	//get data proof
+	respEnv, err = s.GetDataProof(t, worldstate.DefaultDBName, "key1", "admin", rcpt2.GetHeader().GetBaseHeader().GetNumber(), false)
+	require.NoError(t, err)
+	require.NotNil(t, respEnv)
```

Comment:
add a few test cases:
- is deleted, with existing values and deleted ones
- with block numbers that exist but are above or below the actual data's block (have to add a few more blocks with multiple keys for that)
- block that exists, key exists, with data that is not right, etc